### PR TITLE
PUB-1261 SSCS daily list validation schema

### PR DESCRIPTION
**Pull request by joshblackmoor**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/PUB-1261

### Change description ###

Added new validation schema for the SSCS daily list.

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/pip/data/management/models/publication/ListType.java [hunk 1]
```diff
@@ -17,7 +17,8 @@ public enum ListType {
     MAGS_STANDARD_LIST(LocationType.VENUE),
     CIVIL_DAILY_CAUSE_LIST(LocationType.VENUE),
     FAMILY_DAILY_CAUSE_LIST(LocationType.VENUE),
-    CIVIL_AND_FAMILY_DAILY_CAUSE_LIST(LocationType.VENUE);
+    CIVIL_AND_FAMILY_DAILY_CAUSE_LIST(LocationType.VENUE),
+    SSCS_DAILY_LIST(LocationType.VENUE);
```
Comment (ChrisS1512) [inline src/main/java/uk/gov/hmcts/reform/pip/data/management/models/publication/ListType.java:21]: Should this be VENUE? (I'm wondering if it's regional instead)

File: src/test/java/uk/gov/hmcts/reform/pip/data/management/service/schemavalidation/SscsDailyListTest.java [hunk 2]
```diff
@@ -0,0 +1,329 @@
...
+import java.io.InputStream;
+import java.nio.charset.StandardCharsets;
+
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+@SpringBootTest(classes = {Application.class, AzureBlobConfigurationTest.class})
+@ActiveProfiles(profiles = "test")
+@DirtiesContext(classMode = DirtiesContext.ClassMode.AFTER_CLASS)
+@AutoConfigureEmbeddedDatabase(type = AutoConfigureEmbeddedDatabase.DatabaseType.POSTGRES)
+
```
Comment (ChrisS1512) [inline src/test/java/uk/gov/hmcts/reform/pip/data/management/service/schemavalidation/SscsDailyListTest.java:29]: Extra line here :)

File: src/main/resources/schemas/sscs_daily_list.json [hunk 3]
```diff
@@ -0,0 +1,414 @@
...
+      }
+    },
+    "venue": {
+      "$id": "#root/venue",
+      "title": "venue",
+      "type": "object",
+      "required": [
+        "venueName"
+      ],
+      "properties": {
```
Comment (ChrisS1512) [inline src/main/resources/schemas/sscs_daily_list.json:109]: Missing venueContact from here

File: src/main/resources/schemas/sscs_daily_list.json [hunk 4]
```diff
@@ -0,0 +1,414 @@
...
+                "pattern": "^.*$"
+              },
+              "courtHouseContact": {
+                "$id": "#root/courtLists/courtList/courtHouse/courtHouseContact",
+                "description": "The contact for the court house",
+                "title": "Court House Contact",
+                "$ref": "#/definitions/venueContact"
+              },
+              "courtRoom": {
+                "$id": "#root/courtLists/courtRoom",
```
Comment (ChrisS1512) [inline src/main/resources/schemas/sscs_daily_list.json:164]: I'm wondering if court room name is needed here. Worth double checking

**Review by ChrisS1512 — CHANGES_REQUESTED** [review]

File: src/main/java/uk/gov/hmcts/reform/pip/data/management/models/publication/ListType.java (hunk 1 above)
Comment (joshblackmoor) [reply src/main/java/uk/gov/hmcts/reform/pip/data/management/models/publication/ListType.java:21]: Just checked ticket, think it should be region. Updating it.

File: src/main/resources/schemas/sscs_daily_list.json (hunk 4 above)
Comment (joshblackmoor) [reply src/main/resources/schemas/sscs_daily_list.json:171]: Added in :)

File: src/main/resources/schemas/sscs_daily_list.json (hunk 3 above)
Comment (joshblackmoor) [reply src/main/resources/schemas/sscs_daily_list.json:110]: Added in :)

Comment (joshblackmoor) [reply src/main/resources/schemas/sscs_daily_list.json:110]: Added in :)

**Review by ChrisS1512 — APPROVED** [review]

**Review by junaidiqbalmoj — COMMENTED** [review]
As per ticket descriptions, **Court House Name** and **Prosecuting Authority (informant)** are multi selected which means that they should be array.

**Comment by joshblackmoor** [comment]
> As per ticket descriptions, **Court House Name** and **Prosecuting Authority (informant)** are multi selected which means that they should be array.

Changed to an array :)

**Pull request merged**